NickvisionTubeConverter.GNOME/Helpers/RowCallbackData.cs

```diff
@@ -0,0 +1,64 @@
+using NickvisionTubeConverter.GNOME.Controls;
+using System;
+using System.Runtime.InteropServices;
+
+namespace NickvisionTubeConverter.GNOME.Helpers;
+
+/// <summary>
+/// An object to pass to move/delete row callbacks in main window
+/// </summary>
+public class RowCallbackData : IDisposable
+{
+    private bool _disposed;
+
+    /// <summary>
+    /// Row to move/delete
+    /// </summary>
+    public DownloadRow Row { get; init; }
+    /// <summary>
+    /// Box to modify
+    /// </summary>
+    public Gtk.Box Box { get; init; }
+    /// <summary>
+    /// GCHandle to pass to unmanaged code
+    /// </summary>
+    public GCHandle? Handle { get; init; }
```

Comment:
You can get rid of the `?` here. The `?` means "this object could be null" but because you have `Handle = GCHandle.Alloc(this)` in the constructor, we know that this object will NEVER be null, thus the `?` is not needed.

This will allow you to remove the `!` from 363 and 374 in `MainWindow.cs`, because the `!` tells the compiler "I know this object can be null but I'm telling you it's not going to be null". But because we remove the `?` meaning the object can never be null, the `!` makes no sense here.

Reply:
I kept `(RowCallbackData)(GCHandle.FromIntPtr(dataHandle).Target!)` because otherwise there's a warning about casting nullable type to non-nullable, everywhere else `!` is removed